### ISSUE TITLE
Operator in 1.6.0 fails to check RGW ObjectStore when SSL is enabled for RGW

### ISSUE THREAD

**Issue by christianhuening** [rook/rook#7672]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
the rook operator attempts to check the readiness of RGW. This fails if `sslSecretRef` and `SecurePort` are specified in the ObjectStore config CRD. 
**Expected behavior:**
rook-operator also uses the ssl secret provided to connect to RGW to check readiness.
**How to reproduce it (minimal and precise):**

create a CephObjectStore CRD with `sslSecretRef` and `SecurePort` set
**File(s) to submit**:

From ceph operator:
```
E | ceph-object-controller: failed to delete object in bucket. RequestError: send request failed
caused by: Delete "http://rook-ceph-rgw-dev-cluster-s3.rook-ceph.svc:443/rook-ceph-bucket-checker-58f34a41-9f34-43cc-b6df-8595c4a27e42/rookHealthCheckTestObject": EOF
```

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod): 1.6.0
* Storage backend version (e.g. for ceph do `ceph -v`): 15.2.10
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

**Comment by travisn** [comment]
@thotz This should be fixed by #7331 or is that a different issue?

**Comment by thotz** [comment]
> @thotz This should be fixed by #7331 or is that a different issue?

Yeah it should be fixed by above PR

**Comment by leseb** [comment]
Closed by https://github.com/rook/rook/pull/7331. Dup of https://github.com/rook/rook/issues/7288
